allows you to capture pictures by using the volume keys on the device
Thanks to ~FXDemolisher for posting this code on the wiki.
Index: /usr/palm/applications/com.palm.app.camera/app/controllers/capture-assistant.js
===================================================================
--- .orig/usr/palm/applications/com.palm.app.camera/app/controllers/capture-assistant.js
+++ /usr/palm/applications/com.palm.app.camera/app/controllers/capture-assistant.js
@@ -33,7 +33,7 @@ CaptureAssistant = Class.create({
 			this.sublaunch = false;
 			this.defaultFilename = null;
 		}
-		
+		this.volumeKeySubscription = undefined;
 	},
 
 	
@@ -85,6 +85,17 @@ CaptureAssistant = Class.create({
 			var elemC = this.controller.get('sagar_console');
 			elemC.style.display = '';
 		}
+	
+		// listen to volume key events
+		this.volumeKeySubscription = new Mojo.Service.Request(
+			'palm://com.palm.keys/audio', 
+			{
+				method: 'status',
+				parameters: {'subscribe': true},
+				onFailure: function() { Mojo.Log.error("Could not subscribe to volume key events"); },
+				onSuccess: this.handleVolumeKeys.bind(this) 
+			}
+		);
 		
 		llog("CaptureAssistant::setup() finished");
 	},
@@ -174,6 +185,11 @@ CaptureAssistant = Class.create({
 		}
 		
 		this.cameraControl.closeCamera();
+	
+		// clean up listener for volume keys
+		if( this.volumeKeySubscription ) {
+			this.volumeKeySubscription.cancel();
+		}
 	},
 	
 	handleCommand: function(event){
@@ -194,6 +210,14 @@ CaptureAssistant = Class.create({
 			this.cameraControl.capture();
 		}
 	},
+
+	// capture on release of volume keys
+	handleVolumeKeys: function(payload) {
+		// capture when either volume up or down buttons are released
+		if( payload.state === 'up' && ( payload.key === 'volume_up' || payload.key === 'volume_down' ) ) {
+			this.cameraControl.capture();
+		}
+	},
   
 	
 	/**
